src/recordlinker/schemas/pii.py

```diff
@@ -154,8 +200,44 @@ def parse_sex(cls, value):
             elif val in ["f", "female"]:
                 return Sex.FEMALE
             return Sex.UNKNOWN
+    
+    @pydantic.field_validator("ssn", mode="before")
+    def parse_ssn(cls, value):
+        """
+        Parse the ssn string 
+        """
+        if value:
+            val = str(value).strip()
+            pattern = re.compile(r"^\d{3}-\d{2}-\d{4}$")
```

Comment:
For increased flexibility, I think we should allow for values missing the hyphens as well.  We'd want to normalize it of course, for comparisons

Reply:
Gotcha. Also do you think it makes sense to store the SSN int he `XXX-XX-XXXX` format? Otherwise we could just remove everything but the digits and store that as `XXXXXXXXX`.

Reply:
I had a [similar question with DOB](https://github.com/CDCgov/RecordLinker/blob/main/src/recordlinker/schemas/pii.py#L216), as we could optimize storage by removing the '-'.  That optimization is small though, and comes at a readability cost.  I think we should just leave them in for now.